packages/@react-spectrum/s2/src/SegmentedControl.tsx

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaLabelingProps,  DOMRef, FocusEvents, FocusableRef, InputDOMProps, ValueBase} from '@react-types/shared';
+import {Radio, RadioGroup, RadioProps, Provider} from "react-aria-components"
+import {useDOMRef, useFocusableRef} from '@react-spectrum/utils';
+import {createContext, forwardRef, ReactNode, useContext, useRef} from 'react';
+import {focusRing, StyleProps, getAllowedOverrides} from './style-utils' with {type: 'macro'};
+import {style} from '../style/spectrum-theme' with {type: 'macro'};
+import {centerBaseline} from './CenterBaseline';
+import {IconContext} from './Icon';
+import {raw} from '../style/style-macro' with {type: 'macro'};
+
+
+interface SegmentedControlProps extends ValueBase<string|null, string>, InputDOMProps, FocusEvents, StyleProps, AriaLabelingProps {
+  children?: ReactNode,
+  /**
+   * The style of the track.
+   * 
+   * @default 'none'
+   */
+  trackStyle?: 'none' | 'track',
```

Comment:
Probably should have mentioned this before (sorry), but we probably should not support the non-track version for now since it isn't accessible.

Reply:
oh lol it's okay. should we keep the prop still and define it like `trackStyle?: 'track'` so that there isn't as big of a API change later?